ledger/light/handle/set_outgoing_request.go

```diff
@@ -0,0 +1,116 @@
+/*
+ *    Copyright 2019 Insolar Technologies
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package handle
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/insolar/insolar/insolar/flow"
+	"github.com/insolar/insolar/insolar/payload"
+	"github.com/insolar/insolar/insolar/record"
+	"github.com/insolar/insolar/ledger/light/proc"
+	"github.com/pkg/errors"
+)
+
+type SetOutgoingRequest struct {
+	dep     *proc.Dependencies
+	message payload.Meta
+	passed  bool
+}
+
+func NewSetOutgoingRequest(dep *proc.Dependencies, msg payload.Meta, passed bool) *SetOutgoingRequest {
+	return &SetOutgoingRequest{
+		dep:     dep,
+		message: msg,
+		passed:  passed,
+	}
+}
+
+func (s *SetOutgoingRequest) Present(ctx context.Context, f flow.Flow) error {
+	msg := payload.SetOutgoingRequest{}
+	err := msg.Unmarshal(s.message.Payload)
+	if err != nil {
+		return errors.Wrap(err, "failed to unmarshal SetIncomingRequest message")
+	}
+
+	rec := record.Unwrap(&msg.Request)
+	request, ok := rec.(*record.OutgoingRequest)
+	if !ok {
+		return fmt.Errorf("wrong request type: %T", rec)
+	}
+
+	var create = request.CallType == record.CTSaveAsChild || request.CallType == record.CTSaveAsDelegate
+
+	if create {
+		return fmt.Errorf("SetOutgoingRequest can't be a cretion request")
+	}
+
+	return s.setRequest(ctx, msg, request, f)
+}
+
+func (s *SetOutgoingRequest) setRequest(
```

Comment:
Let's move this in the `Present`. Since we don't have the second branch, there is no point for a separate funtion.